examples/posenet/src/PoseNet.js

```diff
@@ -0,0 +1,378 @@
+      	  inputs.push(this._inputTensorId);
+      	}
+      	else{
+      	  inputs.push(this._outputLayer[outputLayerIndex]);
+      	  outputLayerIndex++;
+      	}
+        const data = await getDimensionData("separableConv", this._version, i);
+        // regular depthwise convolution
+        if(this._modelArch[i].rate == 1){
+          dimensionWeights.push(reshape(data.shapeWeights[0]));
+          dimensionWeights.push(reshape(data.shapeWeights[1]));
+          weights.push(new Float32Array(transposeWeights(data.weights[0], data.shapeWeights[0])));
+          weights.push(new Float32Array(transposeWeights(data.weights[1], data.shapeWeights[1])));
+          dimensionBias.push(data.shapeBias[0]);
+          dimensionBias.push(data.shapeBias[1]);
+          bias.push(data.bias[0]);
+          bias.push(data.bias[1]);
+          dimensionOut = this._calculateOutput(dimensionIn, dimensionWeights[0], 
+                                      	       this._modelArch[i]["stride"], "depthwise");
+        }
+        // dilated depthwise convoluton
+        else{
+          const dilationData = dilationWeights(new Float32Array(transposeWeights(data.weights[0], data.shapeWeights[0])),
+					       reshape(data.shapeWeights[0]), this._modelArch[i].rate);
+          dimensionWeights.push(dilationData.dimension);
+          dimensionWeights.push(reshape(data.shapeWeights[1]));
+          weights.push(dilationData.dilationWeights);
+          weights.push(new Float32Array(transposeWeights(data.weights[1], data.shapeWeights[1])));
+          dimensionBias.push(data.shapeBias[0]);
+          dimensionBias.push(data.shapeBias[1]);
+          bias.push(data.bias[0]);
+          bias.push(data.bias[1]);
+          dimensionOut = this._calculateOutput(dimensionIn, reshape(data.shapeWeights[0]), 
+					       this._modelArch[i]["stride"], "depthwise");
+        }
+        dimensionIn = dimensionOut;
+        this._outputLayer.push(this._operandIndex);
+        this._model.addOperand({type:type, dimensions: dimensionOut});
+        this._operandIndex++;
+        dimensionOut = this._calculateOutput(dimensionIn, dimensionWeights[1], 1, "pointwise");
+        dimensionIn = dimensionOut;
+        this._outputLayer.push(this._operandIndex);
+        this._model.addOperand({type:type, dimensions: dimensionOut});
+        this._operandIndex++;
+      }
+      for(let j in dimensionWeights){
+        let tensorId = this._operandIndex++;
+        let tensorTypeWeights = {type: type, dimensions: dimensionWeights[j]};
+        this._model.addOperand(tensorTypeWeights);
+        this._model.setOperandValue(tensorId, weights[j]);
+        inputs.push(tensorId);
+        tensorId = this._operandIndex++;
+        let tensorTypeBias = {type: type, dimensions: dimensionBias[j]};
+        this._model.addOperand(tensorTypeBias);
+        this._model.setOperandValue(tensorId, bias[j]);
+        inputs.push(tensorId);
+        if(this._modelArch[i].convType=="conv2d"){
```

Comment:
there are still indent issues in following `if-else` block. I suggest you check with your editor setting to make sure 1) use spaces for tab; 2) 2 spaces for 1 tab.